File: source_files/haskell/Q1-10-lists/p03.hs
-- Find the K'th element of a list.
-- The first element in the list is number 1.

-- Using prelude functions. 
elementAt :: [a] -> Int -> a
elementAt list nb = list!!(nb - 1)

-- Using list recursion. 
elementAt' :: [a] -> Int -> a
elementAt' (x:_) 1 = x
elementAt' [] _ = error "No element at that position."
elementAt' (_:xs) nb = elementAt' xs (nb - 1)  

File: ben/prolog/P01-P28-lists/p03.pl
% Find the K'th element of a list.
% The first element in the list is number 1.

element_at(X,[X|_],1).
element_at(X,[_|L],N0) :-
    N1 is N0 - 1,
    element_at(X,L,N1). 
